pkg/common/key.go

```diff
@@ -0,0 +1,110 @@
+package common
+
+import (
+	"fmt"
+	"strings"
+
+	guuid "github.com/google/uuid"
+)
+
+const (
+	KEY_DELIMETER = "/"
+	LOCKS         = "_locks"
+	COMMENTS      = "_comments"
+	INTERNAL_DB   = "_"
+)
+
+var prefix string
+
+type Key struct {
+	Prefix    string
+	DBName    string
+	TableName string
+	// the id represents uuid for the rows and id for the comments and locks
+	ID string
+}
+
+func SetPrefix(prf string) {
+	prefix = prf
+}
+
+func GetPrefix() string {
+	return prefix
+}
+
+func ParseKey(keyStr string) (*Key, error) {
+	keyParts := strings.Split(keyStr, KEY_DELIMETER)
+	var retKey Key
+	switch l := len(keyParts); {
+	case l < 3:
+		// the key should contains at least 3 mandatory components: dbName, tableName, uuid
+		return nil, fmt.Errorf("wrong formatting key %s", keyStr)
+	case l == 3:
+		retKey = Key{Prefix: "", DBName: keyParts[l-3], TableName: keyParts[l-2], ID: keyParts[l-1]}
+	default:
+		// length of keyParts > 3
+		prefix := keyParts[0]
+		for i := 1; i < l-3; i++ {
+			prefix = fmt.Sprintf("%s%s%s", prefix, KEY_DELIMETER, keyParts[i])
+		}
+		retKey = Key{Prefix: prefix, DBName: keyParts[l-3], TableName: keyParts[l-2], ID: keyParts[l-1]}
+	}
+	if retKey.DBName == "" || retKey.TableName == "" || retKey.ID == "" {
+		return nil, fmt.Errorf("wrong formatting key %s", keyStr)
+	}
+	if retKey.Prefix != prefix {
+		return nil, fmt.Errorf("wrong key, unmatched prefix %s, %s", retKey.Prefix, prefix)
+	}
+	return &retKey, nil
+}
```

Comment:
you tried to do a micro-optimization (making parts of the name optional) and thus resulting in code which is 10x more complicated. I think the cost out weights the benefits and we should rather stick to a standard representation of paths.

Reply:
It is not a micro-optimization, but supporting different user configuration options. You are talking about 10x more complicated code, but it is a single not too complicated function. 
In your scenario you would require from users to provide ovsdbPrefix, serviceName - and btw none of them cannot contain slashes. Think about explanation about all these, relative to the function above.